### ISSUE TITLE
File exception with New update

### ISSUE THREAD

**Issue by sammeboy635** [microsoft/vscode-cpptools#5184]
Seems to be a issue with the new update where if I try to fopen a .txt file its saying the file is not found. When the same code yesterday was working just fine.

**Comment by sean-mcmanus** [comment]
Our extension doesn't run on .txt files. Is this a VS Code bug? Our C/C++ extension normally only process C/C++ files.

Otherwise, could you provide the version of VS Code and of the C/C++ extension? And can you provide the exact text of the error messages you're seeing?

**Comment by sammeboy635** [comment]
For my C programming class I wrote a program to process .txt files. But when I compile and run in visual studio its giving me a error of "No such file or directory". The way I open the file is with a
fopen("input.txt", "r");
Which this .txt file does exists in the same working directory. I ran the same program using GCC from terminal instead of VS and it runs like it should. If maybe you could give me some insights why Visual studio wouldn't be working and gcc would.
Using:
-Ubuntu: 19.10
-Visual Studio: Version 1.43.2 
-C/C++ 0.27.0-insiders4

Things I have tried:
-Un-installing and re-installing visual studio and extensions
-sudo runnning visual studio.
-Creating the file with a fopen("input.txt","w"); first then closing that FILE* and trying to read it
with another fopen("input.txt","r");

Here is a few photos I took
https://imgur.com/a/jJHJ4UV

The files i'm working with also
[Help.zip](https://github.com/microsoft/vscode-cpptools/files/4378680/Help.zip)


![1](https://user-images.githubusercontent.com/33207323/77495064-e6973e80-6e1d-11ea-93b6-9d236e21e9c1.png)
![2](https://user-images.githubusercontent.com/33207323/77495066-e72fd500-6e1d-11ea-85a7-7dc72ae84add.png)
![3](https://user-images.githubusercontent.com/33207323/77495067-e72fd500-6e1d-11ea-9e1f-820e2634c2d9.png)
![C_C++VER](https://user-images.githubusercontent.com/33207323/77495068-e72fd500-6e1d-11ea-99bb-0596ba1b4d1b.png)
![VS_ver](https://user-images.githubusercontent.com/33207323/77495070-e7c86b80-6e1d-11ea-8aa7-6bfe34543d3b.png)

**Comment by sean-mcmanus** [comment]
If you build your binary from VS Code, does it run correctly from the command line? If so, there is probably some problem with your launch.json.

If you build your binary from the command line, does it run correctly from VS Code. If so, there is probably some problem with your tasks.json.

Can you make sure your cwd settings are correct? If that is wrong, it may not be handling the relative paths correctly, so it won't find the files.

**Comment by sean-mcmanus** [comment]
I'm not able to repro the issue with your repro project. The cwd appears correct. Is the binary getting placed in your ${workspaceFolder}?

You might want to try checking http://www.cplusplus.com/reference/cerrno/errno/ to see why the fopen call returning 0.

**Comment by sammeboy635** [comment]
The file did compile in the right directory and ran from cmd. I was looking at my cwd in my tasks again and it's value was set at "cwd": "${fileDirname}", which I assumed that it would be where the file location is. But when I wrote a program to write where it was currently working at it was "dir: /home/samuel/Desktop/Cprogramming" which is not where the .c file is located in. The C file is compiling and saved at "/home/samuel/Desktop/Cprogramming/HomeWork 8". I did not think to check my cwd, sorry for confusion and thank you so much for the help.
